src/components/light.js

```diff
@@ -46,10 +46,37 @@ module.exports.Component = registerComponent('light', {
       // Light type has not changed. Update light.
       Object.keys(diffData).forEach(function (key) {
         var value = data[key];
-        if (['color', 'groundColor'].indexOf(key) !== -1) {
-          value = new THREE.Color(value);
+
+        switch (key) {
+          case 'color':
+            light.color.set(value);
+            break;
+
+          case 'groundcolor':
+            light.groundColor.set(value);
+            break;
+
+          case 'angle':
+            light.angle = degToRad(value);
+            break;
+
+          case 'target':
+            if (value === null) {
```

Comment:
I'd comment this block and also check that there's a target already set. Don't reset if there's no target.

```js
// Reset target if selector is null.
if (light.target && value === null) {

}
```